docs/best_practices.md

```diff
@@ -0,0 +1,130 @@
+
+This document describes good and recommended practices for using the Notary client.
+It includes recommendations on key management, bootstrapping trust for your repositories,
+delegating trust to other signers. You may wish to refer to the [Getting Started](getting_started.md)
+and [Advanced Usage](advanced_usage.md) documents for more information on some of the
+commands.
+
+## Key Management
+
+There are five primary key roles in Notary. In order of how critical they are to protect, most
+to least, they are:
+
+1. Root
+2. Targets
+3. Delegations
+4. Snapshot
+5. Timestamp
+
+We will address each of these classifications of key one by one in reverse order. The
+Targets and Delegations key will be covered together as they have almost identical
+security profiles and use cases.
+
+### Timestamp Key
+
+It is required that the server manages your Timestamp key. This is a convenience measure
+as it would be impractical to have a human involved in regularly signing the Timestamp.
+
+### Snapshot Key
+
+You may choose to allow the server to manage your Snapshot key. This behaviour is default
+in the Docker Content trust integration. Typically if you are working on your own, and
+publish new content regularly, it is safe for you to manage the Snapshot key yourself.
+If however you use Delegations to collaborate with other contributors, you will want to
+allow the server to sign snapshots.
+
+We consider holding the Timestamp and Snapshot keys in well secured online locations an
+acceptable tradeoff between perfect security and usability.
+
+### Targets and Delegations Keys
+
+While compromise of the Timestamp and Snapshot keys allows an attacker to potentially trick
+new or very out of date users into trusting older versions of the TUF repository, compromise
+of a Delegations key, or the Targets key, allows an attacker to sign in completely arbitrary
+content. Therefore it is important to keep these keys secured. However, as they are also
+required for adding, updating, and deleting content in your Trusted Collection, they are
+more likely to be stored on one or more of your personal computers. The Notary client requests
+a password when saving any keys to persistent storage. This password is used as one component
+in the generation of a symmetric AES 256 key that is used to encrypt your signing key in CBC
+mode. When the signing key is required, you are asked to supply the password so that the
+symmetric can can be regenerated, and the signing key decrypted.
+
+While you may be tempted to mirror SSH common practices and generate different Delegations
+keys for every computer you publish from, the provided key rotation functionalities of TUF
+make this unnecessary. However, there is no security weakness in doing so and if you use
+a large number of different computers, it may be desirable simply from a key distribution
+perspective in that you don't have to update so many systems should one key be compromised. 
```

Comment:
Also, should we mention why targets are higher up in priority than delegations?  (e.g. it is used to sign all the top level delegations?)